actionview/lib/action_view/cache_expiry.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ActionView
+  class CacheExpiry
+    class Executor
+      def initialize(watcher:)
+        @cache_expiry = CacheExpiry.new(watcher: watcher)
+      end
+
+      def before(target)
+        @cache_expiry.clear_cache_if_necessary
+      end
+    end
+
+    def initialize(watcher:)
+      @watched_dirs = nil
+      @watcher_class = watcher
```

Comment:
I find the slight naming differences between watcher/watcher_class and then checker a little confusing. Maybe we could clear them up somehow?

Reply:
👍 Removed `@checker`, now just have `@watcher` and `@watcher_class`